contracts/exchange-forwarder/contracts/src/libs/LibForwarderRichErrors.sol

```diff
@@ -0,0 +1,214 @@
+        hex"b0658a43";
+
+    // bytes4(keccak256("InvalidErc721AmountError(uint256)"))
+    bytes4 internal constant INVALID_ERC721_AMOUNT_ERROR_SELECTOR =
+        0x27ed87bf;
+
+    // solhint-disable func-name-mixedcase
+    function UnregisteredAssetProxyError()
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return UNREGISTERED_ASSET_PROXY_ERROR;
+    }
+
+    function UnsupportedAssetProxyError(
+        bytes4 proxyId
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            UNSUPPORTED_ASSET_PROXY_ERROR_SELECTOR,
+            proxyId
+        );
+    }
+
+    function CompleteFillFailedError()
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return COMPLETE_FILL_FAILED_ERROR;
+    }
+
+    function MakerAssetMismatchError(
+        bytes memory firstOrderMakerAssetData,
+        bytes memory mismatchedMakerAssetData
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            MAKER_ASSET_MISMATCH_ERROR_SELECTOR,
+            firstOrderMakerAssetData,
+            mismatchedMakerAssetData
+        );
+    }
+
+    function UnsupportedFeeError(
+        bytes memory takerFeeAssetData
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            UNSUPPORTED_FEE_ERROR_SELECTOR,
+            takerFeeAssetData
+        );
+    }
+
+    function FeePercentageTooLargeError(
+        uint256 feePercentage
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            FEE_PERCENTAGE_TOO_LARGE_ERROR_SELECTOR,
+            feePercentage
+        );
+    }
+
+    function InsufficientEthForFeeError()
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return INSUFFICIENT_ETH_FOR_FEE_ERROR;
+    }
+
+    function OversoldWethError(
+        uint256 wethSold,
+        uint256 msgValue
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            OVERSOLD_WETH_ERROR_SELECTOR,
+            wethSold,
+            msgValue
+        );
+    }
+
+    function TransferFailedError()
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return TRANSFER_FAILED_ERROR;
+    }
+
+    function DefaultFunctionWethContractOnlyError(
+        address callerAddress
+    )
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodeWithSelector(
+            DEFAULT_FUNCTION_WETH_CONTRACT_ONLY_ERROR_SELECTOR,
+            callerAddress
+        );
+    }
+
+    function InvalidMsgValueError()
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return INVALID_MSG_VALUE_ERROR;
+    }
+
+    function InvalidErc721AmountError(
```

Comment:
Similar to above, can we rename this to something like `Erc721AmountMustEqualOneError`?